src/cgame/cg_particles.cpp

```diff
@@ -2123,9 +2123,8 @@ Calculate the fraction of time passed
 */
 static float CG_CalculateTimeFrac( int birth, int life, int delay )
 {
-	float frac = ( ( float ) cg.time - ( float )( birth + delay ) ) / ( float )( life - delay );
-
-	return Math::Clamp( frac, 0.0f, 1.0f );
+	int diff = life - delay;
+	return diff ? Math::Clamp( (float) (cg.time - birth + delay) / (float) diff, 0.0f, 1.0f ) : 1.0f;
```

Comment:
I wonder if the clamp was a workaround for `infinity` and if it is required.

Reply:
It looks like `( cg.time - birth + delay ) / (float) diff` can be > 1.0. E. g: `life = 10; delay = 3`, start at `cg.time = 0`, then at `cg.time = 8`: `diff = 7; cg.time - birth + delay = 11`, then the fraction is ~1.57.

Reply:
Technically, I think we also don't check that `life - delay > 0`, so it could be negative, but that shouldn't be a valid particle system. Doing that check would also skipping the `diff` check here.

Reply:
We may do that instead?

```c++
return diff > 0 ? Math::Clamp( (float) (cg.time - birth + delay) / (float) diff, 0.0f, 1.0f ) : 1.0f;
```

Reply:
I suppose. I don't think it matters all that much because, again, a particle system where `diff` is < 0 shouldn't be valid anyway and should never spawn.